test/functional/pdo_sqlsrv/pdo_ae_output_param_datetimes.phpt

```diff
@@ -0,0 +1,206 @@
+                        $inputPieces = explode(" ", $inputValues[1]);
+                        $inputValues[1] = $inputPieces[0] . " " . $inputPieces[1] . "." . str_repeat("9", $precision) . " " . $inputPieces[2];
+                    } else if ($dataType == "time") {
+                        $inputValues[0] .= "." . str_repeat("0", $precision);
+                        $inputValues[1] .= "." . str_repeat("9", $precision);
+                    }
+                }
+
+                $type = "$dataType($precision)";
+                trace("\nTesting $type:\n");
+
+                //create and populate table
+                $colMetaArr = array(new ColumnMeta($type, "c_det"), new ColumnMeta($type, "c_rand", null, "randomized"));
+                createTable($conn, $tbname, $colMetaArr);
+
+                $stmt = insertRow($conn, $tbname, array("c_det" => $inputValues[0], "c_rand" => $inputValues[1] ), null, $r);
+                
+                // fetch with PDO::bindParam using a stored procedure
+                dropProc($conn, $spname);
+                $spSql = "CREATE PROCEDURE $spname (
+                                @c_det $type OUTPUT, @c_rand $type OUTPUT ) AS
+                                SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname";
+                $conn->query($spSql);
+            
+                // call stored procedure
+                $outSql = getCallProcSqlPlaceholders($spname, 2);
+                foreach ($pdoParamTypes as $pdoParamType) {
+                    $det = 0;
+                    $rand = 0;
+                    $stmt = $conn->prepare($outSql);
+                    trace("\nParam $pdoParamType with INOUT = $inout\n");
+                    
+                    if ($inout) {
+                        $paramType = $pdoParamType | PDO::PARAM_INPUT_OUTPUT;
+                    } else {
+                        $paramType = $pdoParamType;
+                    }
+                    
+                    $len = 2048;
+                    if ($pdoParamType == PDO::PARAM_BOOL || $pdoParamType == PDO::PARAM_INT) {
+                        $len = PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE;
+                    }
+
+                    $stmt->bindParam(1, $det, $paramType, $len); 
+                    $stmt->bindParam(2, $rand, $paramType, $len); 
+                
+                    try {
+                        $stmt->execute();
+                        $errMsg = "****$type as $pdoParamType failed with INOUT = $inout:****\n";
+                        // What follows only happens with OUTPUT parameter
+                        if ($inout) {
+                            echo "Any datetime type as INOUT param should have caused an exception!\n";
+                        }
+                        if ($pdoParamType == PDO::PARAM_INT) {
+                            // Expect an integer, the first part of the date time string
+                            $ch = ($dataType == "time")? ':' : '-';

```

Comment:
what happens to datetime2 and datetimeoffset

Reply:
are you asking about inout parameter? Binding any date / time types as inout throw exceptions